cocos/2d/components/ui-opacity.ts

```diff
@@ -103,31 +112,41 @@ export class UIOpacity extends Component {
             render.renderEntity.colorDirty = dirty;
             if (uiOp) {
                 render.renderEntity.localOpacity = interruptOpacity * uiOp.opacity / 255;
+                uiOp._setByParent = setByParent;
             } else {
                 // there is a just UIRenderer but no UIOpacity on the node, we should just transport the parentOpacity to the node.
                 render.renderEntity.localOpacity = interruptOpacity;
             }
+            render.node._uiProps.localOpacity = render.renderEntity.localOpacity;
             interruptOpacity = 1;
         } else if (uiOp) {
             // there is a just UIOpacity but no UIRenderer on the node.
             // we should transport the interrupt opacity downward
             interruptOpacity = interruptOpacity * uiOp.opacity / 255;
+            uiOp._setByParent = setByParent;
         }
-
         for (let i = 0; i < node.children.length; i++) {
-            UIOpacity.setEntityLocalOpacityDirtyRecursively(node.children[i], dirty || (interruptOpacity < 1), interruptOpacity);
+            UIOpacity.setEntityLocalOpacityDirtyRecursively(node.children[i], dirty || (interruptOpacity < 1), interruptOpacity, true);
         }
     }
 
     @serializable
     protected _opacity = 255;
 
     public onEnable (): void {
+        // If it has been set by the parent node or grandfather node, it does not need to be initialized.
```

Comment:
Could you please add the reason why it does not need to be initialized if it is set by parents or grandfather?

Reply:
ok